Source/MpCompatAttributes.cs

```diff
@@ -0,0 +1,485 @@
+
+                                if (syncMethod.cancelIfAnyArgNull)
+                                    sync.CancelIfAnyArgNull();
+                                if (syncMethod.cancelIfNoSelectedMapObjects)
+                                    sync.CancelIfNoSelectedMapObjects();
+                                if (syncMethod.cancelIfNoSelectedWorldObjects)
+                                    sync.CancelIfNoSelectedWorldObjects();
+                                if (syncMethod.debugOnly)
+                                    sync.SetDebugOnly();
+                                if (syncMethod.exposeParameters != null)
+                                {
+                                    foreach (var index in syncMethod.exposeParameters)
+                                        sync.ExposeParameter(index);
+                                }
+
+                                break;
+                            }
+                            case MpCompatSyncWorkerAttribute syncWorker:
+                            {
+                                if (RegisterSyncWorker == null)
+                                    continue;
+                                if (method.ReturnType != typeof(void))
+                                    throw new Exception("Cannot register sync worker: return type must be null.");
+                                var parms = method.GetParameters();
+                                if (parms is not { Length: 2 })
+                                    throw new Exception("Cannot register sync worker: delegate doesn't have 2 parameters");
+                                if (parms[0].ParameterType != typeof(SyncWorker))
+                                    throw new Exception($"Cannot register sync worker: the first argument isn't of type {nameof(SyncWorker)}");
+
+                                // The type is normally a ref type, get one that isn't
+                                var genericNonRefType = parms[1].ParameterType.GetElementType();
+                                var delType = typeof(SyncWorkerDelegate<>).MakeGenericType(genericNonRefType);
+                                var del = Delegate.CreateDelegate(delType, method);
+
+                                RegisterSyncWorker.MakeGenericMethod(genericNonRefType)
+                                    .Invoke(null, new object[] { del, syncWorker.Type, syncWorker.isImplicit, syncWorker.shouldConstruct });
+
+                                break;
+                            }
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        Log.Error($"MpCompatPatch {method.DeclaringType}.{method.Name} failed with exception: {e}");
+                    }
+                }
+            }
+
+            foreach (var field in AccessTools.GetDeclaredFields(type).Where(f => f.IsStatic && typeof(ISyncField).IsAssignableFrom(f.FieldType)))
```

Comment:
Can't this be merged above? It's calling the same `AccessTools.GetDeclaredFields(type).Where(m => m.IsStatic)`

Reply:
It's not `GetDeclaredFields` above, but `GetDeclaredMethods`.